src/cmd/flux-watch.py

```diff
@@ -0,0 +1,207 @@
+        metavar="STATE|RESULT",
+        default=set(),
+        help="Watch jobs with specific job state or result",
+    )
+    parser.add_argument(
+        "--since",
+        action=FilterAction,
+        type=str,
+        metavar="WHEN",
+        default=0.0,
+        help="Include only jobs that have been inactive since WHEN. "
+        + "(implies -a if no other --filter option is specified)",
+    )
+    parser.add_argument(
+        "--progress",
+        action="store_true",
+        default=False,
+        help="Show progress bar",
+    )
+    parser.add_argument(
+        "--jps",
+        action="store_true",
+        default=False,
+        help="Display jobs/s instead of elapsed time ",
+    )
+    parser.add_argument(
+        "-l",
+        "--label-io",
+        action="store_true",
+        default=False,
+        help="label lines of output with task id",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Increase verbosity",
+    )
+    parser.add_argument(
+        "jobids",
+        metavar="JOBID",
+        type=JobID,
+        nargs="*",
+        help="Watch specific jobids",
+    )
+    parser.set_defaults(filtered=False)
+    return parser.parse_args()
+
+
+LOGGER = logging.getLogger("flux-watch")
+
+
+@flux.util.CLIMain(LOGGER)
+def main():
+
+    sys.stdout = open(
+        sys.stdout.fileno(), "w", encoding="utf8", errors="surrogateescape"
+    )
+    sys.stderr = open(
+        sys.stderr.fileno(), "w", encoding="utf8", errors="surrogateescape"
+    )
+
+    starttime = None
+    since = 0.0
+
+    args = parse_args()
+
+    if args.jobids and args.filtered:
+        LOGGER.warning("Filtering options ignored with jobid list")
+
+    fh = flux.Flux()
+    instance_starttime = float(fh.attr_get("broker.starttime"))
+
+    if args.since:
+        #  Implies --all unless any other filter option is already in effect.
+        if not args.filter:
+            args.all = True
+        try:
+            since = flux.util.parse_datetime(args.since).timestamp()
+        except ValueError:
+            since = float(args.since)
+
+        # Ensure args.since is in the past
+        if since > flux.util.parse_datetime("now").timestamp():
+            LOGGER.error("--since=%s appears to be in the future", args.since)
+            sys.exit(1)
+
+        #  With since, start elapsed timer at maximum of since time or
+        #  instance starttime
+        starttime = max(since, instance_starttime)
+
+    if args.all:
+        args.filter.update(("pending", "running", "inactive"))
+        if not starttime:
+            starttime = instance_starttime
+
+    if not args.filter:
+        args.filter = {"pending", "running"}
+
+    joblist = JobList(
+        fh,
+        ids=args.jobids,
+        attrs=["state", "result", "t_submit"],
+        filters=args.filter,
+        since=since,
+        user=str(os.getuid()),
+        max_entries=0,
```

Comment:
I can't help but look at the "max_entries = 0" and the "--all" option and think "this will not end well".  Although I can't think of any better alternative, other than perhaps a "be careful what you wish for" in the help/manpage.

Reply:
Yeah, I had the same thought and added a `--count=N` option with default of 1000. I'm still adding tests and fixing things as they come up, but will push a non WIP version shortly.